exporter/containerimage/writer.go

```diff
@@ -114,36 +114,50 @@ func (ic *ImageWriter) Commit(ctx context.Context, inp *exporter.Source, session
 		}
 
 		var ref cache.ImmutableRef
-		var p exptypes.Platform
+		var p *exptypes.Platform
 		if len(ps.Platforms) > 0 {
-			p = ps.Platforms[0]
+			p = &ps.Platforms[0]
 			if r, ok := inp.FindRef(p.ID); ok {
 				ref = r
 			}
 		} else {
 			ref = inp.Ref
 		}
+		config := exptypes.ParseKey(inp.Metadata, exptypes.ExporterImageConfigKey, p)
 
 		remotes, err := ic.exportLayers(ctx, opts.RefCfg, session.NewGroup(sessionID), ref)
 		if err != nil {
 			return nil, err
 		}
+		remote := &remotes[0]
+		if opts.RewriteTimestamp {
+			remote, err = ic.rewriteRemoteWithEpoch(ctx, opts, remote)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Probably these lines didn't need to be moved

Reply:
Likely not, I was just trying to match the structure of the non-map case - I can move it back if it's significant.